test/validations-test.js

```diff
@@ -1,73 +1,66 @@
 const chai = require("chai");
-const sinonChai = require("sinon-chai");
 const expect = chai.expect;
 const Liquid = require("../index.js");
-chai.use(sinonChai);
+const { checkValidDynamicTable } = require("../validations.js");
 
-describe("checks whether valid JSON has been passed to parseAssign statements in liquid expressions", () => {
-  const commonSetup = () => {
-    const engine = Liquid({});
-    return engine;
-  };
+describe("performs validations on liquidJS expressions", () => {
+  before(() => {
+    engine = Liquid({}); // Initialize once before all tests
+  });
 
-  it("should return an empty array if parseAssign is assigned a valid JSON value", () => {
-    const engine = commonSetup();
-    const expression = `
-      {% unless a == "EUR" %}
-        {% parseAssign x4 = "test" %}
-      {% endunless %}
-      {% parseAssign x1 = '{"key": "value"}' %}
-        {% parseAssign x0 = "100" %}
-        {% if a == "USD" %}
-          {% parseAssign x1 = "test" %}
-        {% elsif a == "INR" %}
-          {% parseAssign x2 = "[1, 2, 3]" %}
-        {% else %}
-          {% parseAssign x3 = "true" %}
-        {% endif %}
-      `;
+  describe("dynamic table validations : checks for presence of $$answer = something in each branch of conditional statements and returns error if not found in a branch", () => {
+    it("should return an empty array if there is atleast one occurence of $$answer = something which is not in any if, for or unless block ", () => {
+      const expression = `
+      {% computeColumn testTable test_column %}
+      {% if x %}
+      ...
+      {% elsif y %}
+      ...
+      {% else %}
+      ...
+      {% endif %}
+      {% for item in items %}
+      ...
+      {% endfor %}
+      {% assign $$answer = test %}
```

Comment:
Let's add another case where this `{% assign $$answer = test %}` is above all if conditions instead of being below